### ISSUE TITLE
initSection not working

### ISSUE THREAD

**Issue by baptistebriel** [bigwheel-framework/bigwheel#20]
Hello (again),
I can't figure out how to make the `initSection` work..

```javascript
var bigwheel = require('bigwheel');

module.exports = bigwheel( function(done) {

  done( {
    initSection: require('../sections/Preloader'),
    routes: require('./routes')
  });
});
```
from - https://github.com/bigwheel-framework/documentation/blob/master/routes-special.md#initsection

I think it doesn't pass the `bigwheel.show` function;
```javascript
// check if section is an array or function or object
if(Array.isArray(section)) {

	var sections = [];

	for(var i = 0, len = section.length; i < len; i++) {

		if(typeof section[ i ] == 'object') {

			sections[ i ] = section[ i ];
		} else if(typeof section[ i ] == 'function') {

			sections[ i ] = new section[ i ]();
		}
	}

	this.doShow(viewmediator.apply(undefined, sections), req);
} else if(typeof section == 'object') {
	
	this.doShow(section, req);
} else if(typeof section == 'function') {

	this.doShow(new section(), req);
}

else {
	console.log(section); // undefined
}
```
from - https://github.com/bigwheel-framework/bigwheel/blob/master/index.js#L218

Do you have any ideas? Thanks a lot.

**Comment by njam3** [comment]
The initSection is similar to the other sections with one main difference, a callback is sent in via the constructor that must be called to initialize the framework when all loading is complete. Here is an example of how to use it.

```javascript
function Preloader(onComplete) {
  this.preloaded = onComplete;
}
Preloader.prototype = {
	init: function( req, done ) {
		done();
	},
  	resize: function(w,h) {
  	},
	animateIn: function( req, done ) {
		done();
		// Load all your assets and then call this.preloaded();
		this.preloaded();
	},
	animateOut: function( req, done ) {
		done();
	},
	destroy: function( req, done ) {
		done();
	}
};
module.exports = Preloader;
```

If this isn't working for you, send me a link to you code, and I'll try to help you debug it.

**Comment by baptistebriel** [comment]
Yes, it's the code I'm using.
See – http://baptistebriel.github.io/biggie-initSection v.s http://baptistebriel.github.io/biggie
Thanks for your help!

**Comment by njam3** [comment]
Could provide a link to the unminified version of your code that uses the initSection?

**Comment by baptistebriel** [comment]
Oh sorry, right here - https://github.com/baptistebriel/biggie-initSection

**Comment by njam3** [comment]
lol, i should have been able to figure that our from the first link you sent. I've figured out the issue, seems like it was a version issue. I've fixed it in the latest version of bigwheel. I would suggest removing these entries from your package.json

```
    "bigwheel": "^3.0.1",
    "bw-router": "~1.1.3",
    "bw-viewmediator": "^2.0.0",
    "bw-vm": "^2.0.0",
```

And then doing a fresh ```npm i bigwheel --save```

That should fix the issue, if not let me know and I'll take another look.
